src/util/simplify_expr.cpp

```diff
@@ -405,6 +472,88 @@ static simplify_exprt::resultt<> simplify_string_char_at(
   return c;
 }
 
+/// Take the passed-in constant string array and lower-case every character.
+static void lower_case_string_expression(array_exprt &string_data)
+{
+  auto &operands = string_data.operands();
+  for(auto &operand : operands)
+  {
+    int character = std::stoi(id2string(operand.get(ID_value)), nullptr, 16);
+    if(isalpha(character))
+    {
+      char buffer[33];
+      if(isupper(character))
+      {
+        sprintf(buffer, "%x", tolower(character));
+        operand.set(ID_value, buffer);
+      }
+    }
+  }
+}
+
+/// Simplify String.startsWith function when arguments are constant
+///
+/// \param expr: the expression to simplify
+/// \param ns: namespace
+/// \return: the modified expression or an unchanged expression
+static simplify_exprt::resultt<> simplify_string_equalsignorecase(
+  const function_application_exprt &expr,
+  const namespacet &ns)
+{
+  // We want to get both arguments of any starts-with comparison, and
+  // trace them back to the actual string instance. All variables on the
+  // way must be constant for us to be sure this will work.
+  auto &first_argument = to_string_expr(expr.arguments().at(0));
+  auto &second_argument = to_string_expr(expr.arguments().at(1));
+
+  const auto first_value_opt =
+    try_get_string_data_array(first_argument.content(), ns);
+
+  if(!first_value_opt)
+  {
+    return simplify_exprt::unchanged(expr);
+  }
+
+  array_exprt first_value = first_value_opt->get();
+
+  const auto second_value_opt =
+    try_get_string_data_array(second_argument.content(), ns);
+
+  if(!second_value_opt)
+  {
+    return simplify_exprt::unchanged(expr);
+  }
+
+  array_exprt second_value = second_value_opt->get();
+
+  // Just lower-case both expressions.
+  lower_case_string_expression(first_value);
+  lower_case_string_expression(second_value);
+
+  // test whether second_value is a prefix of first_value
```

Comment:
Given you check the sizes are equal, so this is *not* a prefix check, surely this is just `first_value == second_value`